src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -106,3 +109,63 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
 
     return nodejsLaunchConfig
 }
+
+/**
+ * Recursively searches directory and all nested directories until a TypeScript file is found
+ * @param dir Directory to search
+ * @returns true if at least one TypeScript file exists
+ */
+function hasTypeScriptFilesRecursive(dir:string): boolean {
+    const files = readdirSync(dir)
+
+    for (let i = 0; i < files.length; i++) {
+        const file = files[i];
+        
+        if (file.substr(-3) === '.ts'){
+            return true
+        }
+    
+        if(statSync(path.join(dir, file)).isDirectory()){
+            if(hasTypeScriptFilesRecursive(path.join(dir, file))) {
+                return true
+            }
+        }
+    }        
+    return false
+}
+
+/**
+ * For non-template debug configs (target = code), compile the project
+ * using a temporary default tsconfig.json file.
+ */
+async function compileTypeScript(config: NodejsDebugConfiguration): Promise<void> {
+    // Compile step for TS projects from source code
```

Comment:
Comment is confusing. Can just be deleted, because the docstring explains it fully.